File: SPAnalyze/src/main/java/io/weicools/sp/analyze/MainActivity.java
package io.weicools.sp.analyze;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate (Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    SharedPreferences sp = getSharedPreferences("settings", Context.MODE_PRIVATE);
    SharedPreferences.Editor editor = sp.edit();
    editor.putString("hello", "world");
    editor.putInt("goal", 3000);
    editor.apply();
  }
}

File: app/src/main/java/com/weiwei/practice/ui/main/MainActivity.kt
package com.weiwei.practice.ui.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.weiwei.practice.R
import com.weiwei.practice.TimeRecorder.recordStopTime
import com.weiwei.practice.window.delegate.EdgeInsetDelegate

/**
 * @author weicools
 */
class MainActivity : AppCompatActivity() {
  companion object {
    private const val TAG = "MainActivity_"

    private const val tag_main = "TAG_MAIN"
  }

  override fun onWindowFocusChanged(hasFocus: Boolean) {
    super.onWindowFocusChanged(hasFocus)
    if (hasFocus) {
      recordStopTime()
    }
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    setContentView(R.layout.activity_main)

    EdgeInsetDelegate(this)
      .setNavigationBarColor(0x20000000)
      .start()

    // // 不让 decorView 给状态栏导航栏留白
    // WindowCompat.setDecorFitsSystemWindows(this.window, false)
    //
    // this.window.statusBarColor = Color.TRANSPARENT
    // this.window.navigationBarColor = Color.TRANSPARENT
  }

}